src/fluid/con2prim_robust.hpp

```diff
@@ -53,17 +51,16 @@ class Residual {
   Residual(const Real D, const Real q, const Real bsq, const Real bsq_rpsq,
            const Real rsq, const Real rbsq, const Real v0sq, const Real Ye,
            const singularity::EOS &eos, const fixup::Bounds &bnds, const Real x1,
-           const Real x2, const Real x3)
+           const Real x2, const Real x3, const Real floor_scale_fac)
       : D_(D), q_(q), bsq_(bsq), bsq_rpsq_(bsq_rpsq), rsq_(rsq), rbsq_(rbsq), v0sq_(v0sq),
-        eos_(eos), bounds_(bnds), x1_(x1), x2_(x2), x3_(x3) {
+        eos_(eos), bounds_(bnds), x1_(x1), x2_(x2), x3_(x3),
+        floor_scale_fac_(floor_scale_fac) {
     lambda_[0] = Ye;
     Real garbage = 0.0;
     bounds_.GetFloors(x1_, x2_, x3_, rho_floor_, garbage);
     bounds_.GetCeilings(x1_, x2_, x3_, gam_max_, e_max_);
 
-#if !USE_C2P_ROBUST_FLOORS
-    rho_floor_ = 1.e-20;
-#endif
+    rho_floor_ *= floor_scale_fac_;
```

Comment:
This is the trick that makes the torus happy -- we allow `c2p_robust` to use floors to never fail, but the floors it uses are much smaller than the real floors we subsequently apply in the normal observer frame in `ApplyFloors`.

Reply:
interesting. I suppose that's probably problem dependent... kind of gross.

Reply:
I think it's only as problem-dependent as the actual floors used in `ApplyFloors` since I think the only restriction on the `c2p_robust` floors for torus stability is that they be much smaller than the `ApplyFloors` floors

Reply:
hmm ok.